### PR TITLE
Add hasInflightPackets.

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

Adds `hasInflightPackets` to the keeper. Used in both WriteUpgradeAckChannel and the handler for ChanUpgradeOpen

closes: #3816

### Commit Message / Changelog Entry

```bash
chore: add hasInflightPackets to keeper.
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/04-channel/keeper/keeper.go [hunk 1]
```diff
@@ -570,3 +570,14 @@ func (k Keeper) iterateHashes(ctx sdk.Context, iterator db.Iterator, cb func(por
...
+
+// hasInflightPackets returns true if there are packet commitments stored and
+// false otherwise.
+//
+//lint:ignore U1000 Ignore unused function temporarily for debugging
+func (k Keeper) hasInflightPackets(ctx sdk.Context) bool {
+	iterator := sdk.KVStorePrefixIterator(ctx.KVStore(k.storeKey), []byte(host.KeyPacketCommitmentPrefix))
+	defer sdk.LogDeferred(ctx.Logger(), func() error { return iterator.Close() })
+
+	return iterator.Valid()
```
Comment (DimitrisJim) [inline modules/core/04-channel/keeper/keeper.go:582]: Could re-use `iterateHashes` with an ugly closure but don't think its worth it. Additionally, logging the closing of the iterator as is done elsewhere.

File: modules/core/04-channel/keeper/keeper.go [hunk 2]
```diff
@@ -570,3 +570,14 @@ func (k Keeper) iterateHashes(ctx sdk.Context, iterator db.Iterator, cb func(por
 		}
 	}
 }
+
+// hasInflightPackets returns true if there are packet commitments stored at the specified channel and
```
Comment (DimitrisJim) [inline modules/core/04-channel/keeper/keeper.go:574]: I think its small enough to be tested indirectly by the places that call it but can add it to export_test and write a small test for it.

**Review by chatton — APPROVED** [review]
LGTM! I'm happy with indirect testing.

**Review by charleenfei — APPROVED** [review]
agreed indirect testing is fine, thanks for this :)

**Review by crodriguezvega — APPROVED** [review]

**Pull request merged**